### PR TITLE
Add leaflet-ant-path plugin

### PR DESCRIPTION
**Pull request by davidbrochart**

### CODE REVIEW

File: ipyleaflet/leaflet.py [hunk 1]
```diff
@@ -392,6 +392,25 @@ class Path(VectorLayer):
...
 
 
+class AntPath(VectorLayer):
+    _view_name = Unicode('LeafletAntPathView').tag(sync=True)
+    _model_name = Unicode('LeafletAntPathModel').tag(sync=True)
+
+    locations = List().tag(sync=True)
+
+    # Options
+    use = Unicode('polyline').tag(sync=True, o=True)
```
Comment (martinRenou) [inline ipyleaflet/leaflet.py]: Maybe use an `Enum` so you can validate the value. `Enum(values=['polyline', 'polygon', 'circle'], default_value='polyline')`

File: js/src/layers/AntPath.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+var L = require('../leaflet.js');
+var vectorlayer = require('./VectorLayer.js');
+var LeafletVectorLayerView = vectorlayer.LeafletVectorLayerView;
+var LeafletVectorLayerModel = vectorlayer.LeafletVectorLayerModel;
+var {antPath} = require('leaflet-ant-path');
+
+var LeafletAntPathModel = LeafletVectorLayerModel.extend({
+    defaults: _.extend({}, LeafletVectorLayerModel.prototype.defaults, {
+        _view_name : 'LeafletAntPathView',
+        _model_name : 'LeafletAntPathModel'
```
Comment (martinRenou) [inline js/src/layers/AntPath.js]: You are missing the model default values (use: 'polyline', radius: 10, etc)

File: js/src/layers/AntPath.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    })
+});
+
+var LeafletAntPathView = LeafletVectorLayerView.extend({
+    create_obj: function () {
+        this.obj = antPath(this.model.get('locations'), this.get_ant_options());
+    },
+
+    model_events: function () {
+        LeafletAntPathView.__super__.model_events.apply(this, arguments);
```
Comment (martinRenou) [inline js/src/layers/AntPath.js]: You don't seem to have added support for changing the `locations` dynamically.

File: js/src/layers/AntPath.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        var key;
+        var o = this.model.get('options');
+        for (var i=0; i<o.length; i++) {
+            key = o[i];
+            this.listenTo(this.model, 'change:' + key, function () {
+                this.obj.setStyle(this.get_ant_options());
+            }, this);
+        }
```
Comment (martinRenou) [inline js/src/layers/AntPath.js]: Could this for-loop could be replaced by:

```javascript
this.model.on_some_change(this.model.get('options'), () => {
    this.obj.setStyle(this.get_ant_options());
});
````
?

File: ipyleaflet/leaflet.py (hunk 1 above)
Comment (davidbrochart) [reply ipyleaflet/leaflet.py]: Good point, thanks.

File: js/src/layers/AntPath.js (hunk 2 above)
Comment (davidbrochart) [reply js/src/layers/AntPath.js]: Right!

File: js/src/layers/AntPath.js (hunk 3 above)
Comment (davidbrochart) [reply js/src/layers/AntPath.js]: Indeed.

File: js/src/layers/AntPath.js (hunk 4 above)
Comment (davidbrochart) [reply js/src/layers/AntPath.js]: Yes, that works, thanks!

**Comment by davidbrochart** [comment]
I think so!

**Pull request merged**